core/src/collections.rs

```diff
@@ -137,25 +152,41 @@ impl MongoStatement for MongoCollections {
             self.current_database_index = Some(0);
         }
         loop {
-            if self
+            while self
```

Comment:
I'm not sure why this is a `while` now instead of an `if`. Can you tell me a case where we need more than one iteration not already covered by the `loop`? Maybe the `loop` can be removed?

Reply:
The`loop` is for moving to the next database, the `while` is for moving to the next collection if the current one does not match the filter. We do need both of them because of the client-side filtering. Does that make sense?

Reply:
Yep! thanks!